icu4c/source/test/intltest/intltest.cpp

```diff
@@ -2091,12 +2084,22 @@ UBool IntlTest::assertEquals(const char* message, std::u16string_view expected,
         possibleDataError);
 }
 
-UBool IntlTest::assertEquals(const char* message,
-                             int32_t expected,
-                             int32_t actual) {
+bool IntlTest::assertSigned64Equals(const char *message, int64_t expected, int64_t actual) {
     if (expected != actual) {
-        errln(UnicodeString("FAIL: ") + message + "; got " +
-              actual + "=0x" + toHex(actual) +
+        errln(UnicodeString("FAIL: ") + message + "; got " + actual + "; expected " + expected);
```

Comment:
no hex?

Reply:
See also the comments in the .h.

Reply:
I did know where we were ;-)
I am just wondering why we show hex for int32_t but only decimal for int64_t.
Sometimes large values would also be more readable in hex, e.g., for collation elements.

Not required here, but maybe consider adding hex output here?

Reply:
The idea was that showing hex gets a bit spammy in general, e.g., when comparing sizes.

But if we think we want hex always, we can simplify the code drastically, by dropping the int32_t overload entirely.

Reply:
Hmm, wait, no, it gets messy.  This 64-bit overload gets used for all numeric types.
If we have negative numbers, we don’t want to log the hex of the unsigned representation of their widening.
(That is, if we have an int8_t that is -1, we want to show FF, not FFFFFFFF; the latter would be massively confusing if you are debugging some byte-twiddling code.)

So this would require lifting some kind of debugString out of the assert functions as I wrote in the other comment, which is fine, but then it would require templatizing it so it doesn’t widen. That gets messy because we cannot use UnicodeString in the header.

Let’s not go there.

Reply:
> That gets messy because we cannot use UnicodeString in the header.

OK, admittedly we don’t _need_ to use UnicodeString here, std::string would do the job just fine for printing out integers.

It would be a nontrivial rewrite though, and we have stuff on top of this. How about doing it in a future PR on the same ticket?

Reply:
> It would be a nontrivial rewrite though, and we have stuff on top of this. How about doing it in a future PR on the same ticket?

ok